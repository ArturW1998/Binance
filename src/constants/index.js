export const ACTION_TYPES = {
  UPDATE_MARKET_PAIRS: 'UPDATE_MARKET_PAIRS',
  SET_ACTIVE_MARKET: 'SET_ACTIVE_MARKET',
  TOGGLE_SOCKET_STREAMS: 'TOGGLE_SOCKET_STREAMS',
};

export const PATHS = {
  BASE_PATH: 'wss://stream.binance.com',
  STREAM_PATH: '/stream',
  STREAM_PARAM: 'streams=',
};

export const LISTS = {
  PAIRS_LIST: ['BNB', 'BTC', 'ETH', 'USDT'],
  DOTS_LIST: [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }],
  TABLE_DATA_LIST: [
    { title: 'Pair' },
    { title: 'Latest Price' },
    { title: 'Open' },
    { title: 'High' },
    { title: 'Low' },
    { title: 'Volume' },
  ],
};

export const CONFIG = {
  BASE_NAME: '/Binance/',
  LOCALSTORAGE_KEY: 'markets',
};
