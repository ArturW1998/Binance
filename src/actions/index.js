import { SET_ACTIVE_MARKET, UPDATE_MARKET_PAIRS, TOGGLE_SOCKET_STREAMS } from '../constants';

export const setActiveMarket = data => ({ type: SET_ACTIVE_MARKET, data });

export const updateMarketPairs = data => ({ type: UPDATE_MARKET_PAIRS, data });

export const toggleSocketStreams = status => ({ type: TOGGLE_SOCKET_STREAMS, status });
