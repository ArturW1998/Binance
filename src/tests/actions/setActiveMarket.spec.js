import { SET_ACTIVE_MARKET } from '../../constants';
import { setActiveMarket } from '../../actions';

describe('setActiveMarket action', () => {
  const expectedActionSetActiveMarket = {
    type: SET_ACTIVE_MARKET,
    data: {
      market: 'BTC',
    },
  };

  const { data } = expectedActionSetActiveMarket;

  it('setActiveMarket', () => {
    expect(setActiveMarket(data)).toEqual(expectedActionSetActiveMarket);
  });

  it('setActiveMarket should return an action', () => {
    expect(setActiveMarket(data)).toMatchSnapshot();
  });
});
