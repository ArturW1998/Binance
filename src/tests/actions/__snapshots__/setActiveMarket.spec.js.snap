// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setActiveMarket action should handle setActiveMarket 1`] = `
Object {
  "data": Object {
    "market": "BTC",
  },
  "type": "SET_ACTIVE_MARKET",
}
`;
